Flaky Test: TestDeleteRunningContainer

https://travis-ci.org/containerd/containerd/jobs/353160768
```
--- FAIL: TestDeleteRunningContainer (0.29s)
	container_test.go:555: failed to mount rootfs component &{overlay overlay [workdir=/var/lib/containerd-test/io.containerd.snapshotter.v1.overlayfs/snapshots/124/work upperdir=/var/lib/containerd-test/io.containerd.snapshotter.v1.overlayfs/snapshots/124/fs lowerdir=/var/lib/containerd-test/io.containerd.snapshotter.v1.overlayfs/snapshots/1/fs]}: no such file or directory: unknown
```

Another similar but not exactly the same failure:
```
--- FAIL: TestSnapshotterClient/MoveFileFromLowerLayer (0.01s)
    	testsuite.go:820: Check snapshots failed: no such file or directory
    		failed to mount
    		github.com/containerd/containerd/snapshots/testsuite.applyToMounts
    			/home/travis/gopath/src/github.com/containerd/containerd/snapshots/testsuite/helpers.go:40
    		github.com/containerd/containerd/snapshots/testsuite.createSnapshot
    			/home/travis/gopath/src/github.com/containerd/containerd/snapshots/testsuite/helpers.go:62
    		github.com/containerd/containerd/snapshots/testsuite.checkSnapshots
    			/home/travis/gopath/src/github.com/containerd/containerd/snapshots/testsuite/helpers.go:123
    		github.com/containerd/containerd/snapshots/testsuite.checkFileFromLowerLayer
    			/home/travis/gopath/src/github.com/containerd/containerd/snapshots/testsuite/testsuite.go:819
    		github.com/containerd/containerd/snapshots/testsuite.makeTest.func1
    			/home/travis/gopath/src/github.com/containerd/containerd/snapshots/testsuite/testsuite.go:108
    		testing.tRunner
    			/home/travis/.gimme/versions/go1.10.linux.amd64/src/testing/testing.go:777
    		runtime.goexit
    			/home/travis/.gimme/versions/go1.10.linux.amd64/src/runtime/asm_amd64.s:2361
    		failed to apply
    		github.com/containerd/containerd/snapshots/testsuite.createSnapshot
    			/home/travis/gopath/src/github.com/containerd/containerd/snapshots/testsuite/helpers.go:63
    		github.com/containerd/containerd/snapshots/testsuite.checkSnapshots
    			/home/travis/gopath/src/github.com/containerd/containerd/snapshots/testsuite/helpers.go:123
    		github.com/containerd/containerd/snapshots/testsuite.checkFileFromLowerLayer
    			/home/travis/gopath/src/github.com/containerd/containerd/snapshots/testsuite/testsuite.go:819
    		github.com/containerd/containerd/snapshots/testsuite.makeTest.func1
    			/home/travis/gopath/src/github.com/containerd/containerd/snapshots/testsuite/testsuite.go:108
    		testing.tRunner
    			/home/travis/.gimme/versions/go1.10.linux.amd64/src/testing/testing.go:777
    		runtime.goexit
    			/home/travis/.gimme/versions/go1.10.linux.amd64/src/runtime/asm_amd64.s:2361
    		failed to create snapshot 1
    		github.com/containerd/containerd/snapshots/testsuite.checkSnapshots
    			/home/travis/gopath/src/github.com/containerd/containerd/snapshots/testsuite/helpers.go:125
    		github.com/containerd/containerd/snapshots/testsuite.checkFileFromLowerLayer
    			/home/travis/gopath/src/github.com/containerd/containerd/snapshots/testsuite/testsuite.go:819
    		github.com/containerd/containerd/snapshots/testsuite.makeTest.func1
    			/home/travis/gopath/src/github.com/containerd/containerd/snapshots/testsuite/testsuite.go:108
    		testing.tRunner
    			/home/travis/.gimme/versions/go1.10.linux.amd64/src/testing/testing.go:777
    		runtime.goexit
    			/home/travis/.gimme/versions/go1.10.linux.amd64/src/runtime/asm_amd64.s:2361
    	helpers.go:67: drwx------       4096 /tmp/snapshot-suite-SnapshotterClient-196955645
    	helpers.go:67: drwxr-xr-x       4096 /tmp/snapshot-suite-SnapshotterClient-196955645/root
    	helpers.go:67: drwxr-xr-x       4096 /tmp/snapshot-suite-SnapshotterClient-196955645/work
```

--- a/snapshots/overlay/overlay.go
+++ b/snapshots/overlay/overlay.go
@@ -309,7 +309,9 @@ func (o *snapshotter) Cleanup(ctx context.Context) error {
 }
 
 func (o *snapshotter) cleanupDirectories(ctx context.Context) ([]string, error) {
-	ctx, t, err := o.ms.TransactionContext(ctx, false)
+	// Get a write transaction to ensure no other write transaction can be entered
+	// while the cleanup is scanning.
+	ctx, t, err := o.ms.TransactionContext(ctx, true)
 	if err != nil {
 		return nil, err
 	}

